### PR TITLE
Added UI changes related to Key Manager usage implementations

### PR DESCRIPTION
**Pull request by SavinduDimal**
### Purpose

- Port https://github.com/wso2-support/apim-apps/pull/157
- Related PRs: https://github.com/wso2/carbon-apimgt/pull/12345
- Related Issues: https://github.com/wso2/api-manager/issues/1943
- Add frontend changes for key manager usages in admin portal
- Add frontend changes for key delete in dev portal

#### Admin portal changes

- Add new column to view key manager usages.

Screenshot 2024-03-22 at 12 23 29

- Add a pop up to view API usages and Application usages of a key manager
Screenshot 2024-03-22 at 12 24 24

Screenshot 2024-03-22 at 12 24 40

#### Dev portal changes

- Add a button to remove keys
Screenshot 2024-03-22 at 12 25 43

### CODE REVIEW

File: portals/admin/src/main/webapp/source/src/app/data/api.js [hunk 1]
```diff
@@ -868,6 +868,42 @@ class API extends Resource {
...
+            );
+        });
+    }
+    
+     /**
+     * Get key manager application usages
+     */
+     getKeyManagerApplicationUsages(keyManagerId) {
+        return this.client.then((client) => {
+            return client.apis["Key Manager (Individual)"].get_key_managers__keyManagerId__app_usages(
```
Comment (lasanthaS) [inline portals/admin/src/main/webapp/source/src/app/data/api.js:888]: Put single quotes here

**Review by lasanthaS — CHANGES_REQUESTED** [review]

File: portals/admin/src/main/webapp/source/src/app/components/KeyManagers/ListKeyManagerUsages.jsx [hunk 2]
```diff
@@ -0,0 +1,485 @@
+/*
+ * Copyright (c) 2023, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (SavinduDimal) [reply portals/admin/src/main/webapp/source/src/app/components/KeyManagers/ListKeyManagerUsages.jsx]: Resolved by https://github.com/wso2/apim-apps/pull/622/commits/04b8ced43a5ae7fea10a867841e7e6e403c77a94

File: portals/admin/src/main/webapp/source/src/app/data/api.js [hunk 3]
```diff
@@ -868,6 +868,42 @@ class API extends Resource {
...
+            );
+        });
+    }
+
+     /**
+     * Get key manager api usages
+     */
+     getKeyManagerApiUsages(keyManagerId) {
+        return this.client.then((client) => {
+            return client.apis["Key Manager (Individual)"].get_key_managers__keyManagerId__api_usages(
```
[the remark this answers is not in the document]
Comment (SavinduDimal) [reply portals/admin/src/main/webapp/source/src/app/data/api.js]: Resolved by https://github.com/wso2/apim-apps/pull/622/commits/04b8ced43a5ae7fea10a867841e7e6e403c77a94

File: portals/admin/src/main/webapp/source/src/app/data/api.js (hunk 1 above)
Comment (SavinduDimal) [reply portals/admin/src/main/webapp/source/src/app/data/api.js]: Resolved by https://github.com/wso2/apim-apps/pull/622/commits/04b8ced43a5ae7fea10a867841e7e6e403c77a94

**Review by lasanthaS — APPROVED** [review]

**Pull request merged**